guacamole/src/main/java/org/apache/guacamole/rest/APIRequest.java

```diff
@@ -58,6 +68,18 @@ public APIRequest(HttpServletRequest request,
 
         super(request);
 
+        // Grab the remote host info.
+        if (request.getRemoteHost() != null && !request.getRemoteHost().isEmpty())
```

Comment:
Removed the unnecessary checks and just assigned both remote host and remote addr.